src/main/java/agh/edu/pl/MapNotes/controller/UserController.java

```diff
@@ -0,0 +1,90 @@
+package agh.edu.pl.MapNotes.controller;
+
+import agh.edu.pl.MapNotes.exception.UserNotFoundException;
+import agh.edu.pl.MapNotes.model.User;
+import agh.edu.pl.MapNotes.model.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.HashMap;
+import java.util.List;
+
+
+@RestController
+@RequestMapping("/user")
+public class UserController {
+    private final UserRepository userRepository;
+
+    @Autowired
+    UserController(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    /**
+     * Get all users in the database.
+     * @return list of all users in the database.
+     */
+    @GetMapping
+    public List<User> getAllUsers() {
+        return userRepository.findAll();
+    }
+
+    /**
+     * Update or create map.
+     * Function allow to create user or to update.
+     * If the id match id of any user it will update that user.
+     * Otherwise it will create new user with new id.
+     * @param user map to update or create.
+     * @return map stored in database after operation.
+     */
+    @PutMapping
+    public User putUser(@Valid @RequestBody User user) {
+        user = userRepository.save(user);
+        return this.userRepository.findById(user.getId()).get();
+    }
+
+
+    @RequestMapping("/addIfNotExist")
+    public Boolean putUserIfNotInDatabase(@RequestBody User user) {
+        if (this.isUserCredentialsRight(user)) {
+            return false;
+        }
+        userRepository.save(user);
+        return true;
+    }
+
+    /**
+     * Get information about specified map.
+     * @param userId id of map to get.
+     * @return founded user.
+     * @throws UserNotFoundException when user was not found in database.
+     */
+    @GetMapping("/{userId}")
+    public User getUserById(@PathVariable("userId") Long userId) throws UserNotFoundException {
+        return userRepository.findById(userId).orElseThrow(UserNotFoundException::new);
+    }
+
+    @RequestMapping("/exists")
+    public Boolean isUserCredentialsRight (@RequestBody User user) {
```

Comment:
Nie wiem czy jest sens taką funkcje umieszczać jako API. Walidacja na froncie powina nie wymuszać co chwile zapytywania backendu, a jedynie ostateczna akcja powinna sprawdzać. Ponad to to co tutaj sprawdzacie nie jest rolą controllera tylko bardziej modelu/bazy danych. W szczególności że używacie tutaj funkcji podczas tworzenia użytkownika ale już podczas updateu nie.